src/csp_io.c

```
@@ -122,42 +122,46 @@ void csp_id_copy(csp_id_t * target, csp_id_t * source) {
 
 int csp_send_direct(csp_id_t idout, csp_packet_t * packet, int from_me) {
 
-	int ret;
-
 	/* Try to find the destination on any local subnets */
 	int via = CSP_NO_VIA_ADDRESS;
-	csp_iface_t * iface = csp_iflist_get_by_subnet(idout.dst);
+	csp_iface_t * iface = NULL;
+	csp_packet_t * copy = NULL;
+	int local_found = 0;
 
-	/* Otherwise, resort to the routing table for help */		
-	if (iface == NULL) {
-		csp_route_t * route = csp_rtable_find_route(idout.dst);
-		if (route == NULL) {
-			csp_dbg_conn_noroute++;
-			return CSP_ERR_TX;
+	while ((iface = csp_iflist_get_by_subnet(idout.dst, iface)) != NULL) {
+		
+		/* Apply outgoing interface address to packet */
+		idout.src = iface->addr;
+		
+		copy = csp_buffer_clone(packet);
+		if (csp_send_direct_iface(idout, copy, iface, via, 1) != CSP_ERR_NONE) {
+			csp_buffer_free(copy);
```

When you failed to send, don't you wanna tell caller what happened?

---

Nope. See above.

The sender reacting to errors is opening a large can of trouble to the sender. And it won't actually ensure any better performance. Because the packet could be silently dropped anywhere in the network anyway.